src/rdsn/src/common/backup_restore_common.h

```diff
@@ -29,6 +29,7 @@ namespace dsn {
 namespace replication {
 
 DSN_DECLARE_string(cold_backup_root);
+DSN_DECLARE_uint32(cold_backup_checkpoint_reserve_minutes);
```

Comment:
It's no needed to declare it in header files, declare it in the cpp files where you use it is OK?

Reply:
It will be used in different cpp files, so I think it need to be defined in header file.

Reply:
Larger header file will cause build slower

Reply:
Maybe other configuration can declare in cpp file, but as my comment this config `cold_backup_checkpoint_reserve_minutes` will be used in different cpp files, it need to be defined in header file.

Reply:
If I disn't miss anything, ther are only 2 files use it, delare it in thw 2 files will cause any problem?

Reply:
In my memory, it will be used in at least 3 files. Will declaring it in different 3 cpp files is more effecitive than declaring it in one header file? I just don't know which way is more effective.

Reply:
I think it's a type of combine rule of Include_What_You_Use and Forward_Declarations. If the header file dose use the header/type itself, avoid including / declearing it in header files, include or declear it in the place where really use it.

1. https://google.github.io/styleguide/cppguide.html#Include_What_You_Use
2. https://google.github.io/styleguide/cppguide.html#Forward_Declarations

Reply:
Got it~ Updated :-)